src/compiler/csharp_generator.cc

```diff
@@ -41,6 +38,9 @@ using grpc_generator::METHODTYPE_CLIENT_STREAMING;
 using grpc_generator::METHODTYPE_NO_STREAMING;
 using grpc_generator::METHODTYPE_SERVER_STREAMING;
 using grpc_generator::StringReplace;
+using proto2::compiler::csharp::GetClassName;
```

Comment:
you'd need to put in place some kind of switching between the internal and external namespaces (based on which version of protobuf_config.h gets used). 
So far you've tried to use the external-only namespace names internally (which would break the internal build) and now you're trying to use the internal-only namespaces externally and that won't work either, for the same reason.

If you look at  
https://github.com/grpc/grpc/blob/master/src/compiler/config_protobuf.h 

and https://github.com/grpc/grpc/blob/86f93801e5d452cff40d6958d21a8eb211da0654/include/grpcpp/impl/codegen/config_protobuf.h 
you will see that it defines a bunch of macros that allow you to change  namespaces/types of objects being used in the generator plugin code. You need to use such technique to allow calling e.g `google::protobuf::compiler::csharp::GetClassName` externally and `proto2::compiler::csharp::GetClassName` (or whatever) internally.

Reply:
Got it, update added